terraform/open-vpn/ec2.tf

```diff
@@ -0,0 +1,41 @@
+resource "tls_private_key" "open_vpn_key_pair_config" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "aws_key_pair" "generated_open_vpn_server_keypair" {
+  key_name   = var.open_vpn_key_pair_name
+  public_key = tls_private_key.open_vpn_key_pair_config.public_key_openssh
+}
+
+resource "aws_instance" "open_vpn_ec2" {
+  ami                    = "ami-05c424d59413a2876"
```

Comment:
should be using a query to find the AMI id, unless you need to pin it to a specific version of the AMI?

Reply:
Or pass it in as a variable.

Reply:
specific versions. the scripts dont work on older 18 versions of ubuntu. 
Because of how the silent installs work and the "assumptions" that the silent --unattended installs work it needs to be maintained away from a consumers control in order to maintain "out the box working"

Reply:
Well at least pass it in as a parameter (default to the current working AMI ID) so it is flexible to change if required?

Reply:
Just don't like hard coded IDs in the TF files.

Reply:
that makes sense.. i will variablise with defaults and description. that sounds like a gd plan